frontend/src/metabase/collections/components/CollectionContent/CollectionItemsTable.tsx

```diff
@@ -0,0 +1,200 @@
+  CollectionId,
+  CollectionItem,
+  CollectionItemModel,
+} from "metabase-types/api";
+import { SortDirection, type SortingOptions } from "metabase-types/api/sorting";
+
+import {
+  CollectionEmptyContent,
+  CollectionTable,
+} from "./CollectionContent.styled";
+
+export type CollectionItemsTableProps = {
+  collectionId: CollectionId;
+} & Partial<{
+  bookmarks: Bookmark[];
+  clear: () => void;
+  collection: Collection;
+  createBookmark: CreateBookmark;
+  databases: Database[];
+  deleteBookmark: DeleteBookmark;
+  getIsSelected: (item: CollectionItem) => boolean;
+  handleCopy: (items: CollectionItem[]) => void;
+  handleMove: (items: CollectionItem[]) => void;
+  hasPinnedItems: boolean;
+  loadingPinnedItems: boolean;
+  models: CollectionItemModel[];
+  pageSize: number;
+  selectOnlyTheseItems: (items: CollectionItem[]) => void;
+  selected: CollectionItem[];
+  toggleItem: (item: CollectionItem) => void;
+  onClick: (item: CollectionItem) => void;
+  showActionMenu: boolean;
+}>;
+
+export const CollectionItemsTable = ({
+  collectionId,
+  collection,
+  getIsSelected,
+  selectOnlyTheseItems,
+  databases,
+  bookmarks,
+  createBookmark,
+  deleteBookmark,
+  loadingPinnedItems,
+  hasPinnedItems,
+  selected,
+  toggleItem,
+  clear,
+  handleMove,
+  handleCopy,
+  pageSize = COLLECTION_PAGE_SIZE,
+  models = ALL_MODELS,
+  onClick,
+  showActionMenu = true,
+}: CollectionItemsTableProps) => {
+  const [unpinnedItemsSorting, setUnpinnedItemsSorting] =
+    useState<SortingOptions>({
+      sort_column: "name",
+      sort_direction: SortDirection.Asc,
+    });
+
+  const [total, setTotal] = useState<number | null>(null);
+
+  const { handleNextPage, handlePreviousPage, setPage, page, resetPage } =
+    usePagination();
+
+  useEffect(() => {
+    if (collectionId) {
+      resetPage();
+      setTotal(null);
+    }
+  }, [collectionId, resetPage]);
+
+  const handleUnpinnedItemsSortingChange = useCallback(
+    (sortingOpts: SortingOptions) => {
+      setUnpinnedItemsSorting(sortingOpts);
+      setPage(0);
+    },
+    [setPage],
+  );
+
+  const showAllItems = !collection || isRootTrashCollection(collection);
+
+  const unpinnedQuery = {
+    collection: collectionId,
+    models,
+    limit: pageSize,
+    offset: pageSize * page,
+    ...(showAllItems ? {} : { pinned_state: "is_not_pinned" }),
+    ...unpinnedItemsSorting,
+  };
+
+  return (
+    <Search.ListLoader
+      query={unpinnedQuery}
+      loadingAndErrorWrapper={false}
+      keepListWhileLoading
+      wrapped
+    >
+      {({
+        list: unpinnedItems = [],
+        metadata = {},
+        loading: loadingUnpinnedItems,
+      }: {
+        list: CollectionItem[];
+        metadata: { total?: number };
+        loading: boolean;
+      }) => {
+        if (metadata.total) {
+          setTotal(metadata.total);
+        }
+
+        const hasPagination: boolean = total ? total > pageSize : false;
+
+        const unselected = getIsSelected
```

Comment:
This is another place where we have an SDK-specific logic.

I'm not sure if it'd be a better idea to have a different branch entirely, so we don't have to ensure all these props exist because in CollectionBrowser, we only send a handful of props, and some of these props don't even make sense e.g. We can't bulk select items in CollectionBrowser.

I'd imagine the component could look more like this
```ts
const isEmbeddingSdk = useSelector(getIsEmbeddingSdk)

if (isEmbeddingSdk) {
  // render the component with relevant props, and omit any props if we don't need them.
  return <CollectionTable data-testid="collection-table">...
}

// render the component like before. Here we don't have to have any SDK specific logic apply to each prop.
return <CollectionTable data-testid="collection-table">...
```

Reply:
Yeah, I think if I had to make more changes this would have already become its own component. A lot of the components that are lower in this component chain already accepted a bunch of optional parameters (and checked for them as well) so it wasn't a huge change in terms of adding boolean checks and things like that

Reply:
I think for now we should keep it and talk about splitting in another PR - if we branch here I think we could probably find a component lower in the chain that we can use in `CollectionBrowser`